homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,85 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height> arr) {
+    std::vector<std::pair<uint8_t, uint8_t>> vec;
+    uint8_t color = 0; // color of current character
+    uint8_t lastcolor = 0; // color of current character 
+    unsigned int sameColorCount = 0; // sameColorCount counts how many neighbouring characters have the same color 
+    // for loop to iterate through all characters in the array arr, x, y - character coordinates
+    for (int y = 0; y < height; y++) {
+        for (int x = 0; x < width; x++) {
+            if (x == 0) {
+                
+                lastcolor = arr[y][x];
+                sameColorCount = 1;
+            }
+            else { 
+                color = arr[y][x];
+
+                if (color == lastcolor) {
+                    sameColorCount++;
+                }
+                else {
+                    vec.push_back(std::make_pair(lastcolor, sameColorCount));
+                    lastcolor = color;
+                    sameColorCount = 1;
+                }
+                if(x == width - 1) {
+                    vec.push_back(std::make_pair(lastcolor, sameColorCount)); 
+                    sameColorCount = 1;       
+                }
+            }
```

Comment:
You can simplify it. Just increment counter and when 2 colors ar different push them on vector and start counting from one. This will reduce amount of code :)